packages/flutter_tools/lib/src/commands/create.dart

```diff
@@ -491,21 +503,59 @@ To edit platform code in an IDE see https://flutter.dev/developing-packages/#edi
 void _printPluginUpdatePubspecMessage(String pluginPath, String platformsString) {
   globals.printStatus('''
 You need to update $pluginPath/pubspec.yaml to support $platformsString.
-
 ''', emphasis: true, color: TerminalColor.red);
 }
 
 void _printNoPluginMessage() {
     globals.printError('''
 You've created a plugin project that doesn't yet support any platforms.
-
 ''');
 }
 
-void _printAddPlatformMessage(String pluginPath) {
+void _printPluginAddPlatformMessage(String pluginPath) {
   globals.printStatus('''
 To add platforms, run `flutter create -t plugin --platforms <platforms> .` under $pluginPath.
 For more information, see https://flutter.dev/go/plugin-platforms.
 
 ''');
 }
+
+// returns a list disabled, but requested platforms
+List<String> _getPlatformWarningList(List<String> requestedPlatforms) {
+  final List<String> platformsToWarn = <String>[
+  if (requestedPlatforms.contains('web') && !featureFlags.isWebEnabled)
+    'web',
+  if (requestedPlatforms.contains('macos') && !featureFlags.isMacOSEnabled)
+    'macos',
+  if (requestedPlatforms.contains('windows') && !featureFlags.isWindowsEnabled)
+    'windows',
+  if (requestedPlatforms.contains('linux') && !featureFlags.isLinuxEnabled)
+    'linux',
+  ];
+
+  return platformsToWarn;
+}
+
+void _printWarningDisabledPlatform(List<String> platforms) {
+  final  List<String> desktop = <String>[], web = <String>[];
+  for (final String platform in platforms) {
+    if (platform == 'web') {
+      web.add(platform);
+    } else if (platform == 'macos' || platform == 'windows' || platform == 'linux') {
+      desktop.add(platform);
+    }
+  }
+
+  if (desktop.isNotEmpty) {
+    globals.printStatus('''
```

Comment:
this line is too long, lets break this up into more statements, for example: pull out this computations

final String verb = desktop.length > 1 ? 'are' : 'is';
final String platforms = desktop.length > 1 ? 'platforms' : 'platform';
...

Reply:
Done